### DISCUSSION TITLE
Jellyfin 豆瓣元数据插件

### DISCUSSION THREAD

**Discussion by giscus-xzonn** [Xzonn/xzonn.github.io#22] [General]
# Jellyfin 豆瓣元数据插件

Jellyfin 豆瓣元数据插件 - 来自 Xzonn 的小站

https://xzonn.top/posts/Jellyfin-Plugin-Douban.html

**Comment by 15320179666** [comment]
感谢UP主的分享。请教一个问题：我按照你的说法设置好后，重新刷新元数据，目前看元数据没有变化，请问这是哪里出的问题？

**Reply by Xzonn** [reply]
新安装之后需要编辑之前存在的媒体库的设置，选择豆瓣作为元数据来源。

**Reply by 15320179666** [reply]
按照UP主的补充说明设置后，开始削刮海报墙了。感谢。

**Comment by sxjsjt** [comment]
你好，我Windows版Jellyfin添加插件后，显示NotSupported，Jellyfin版本为10.8.9。请问这是什么问题？

**Reply by Xzonn** [reply]
官网已经无法下载旧版本，尝试升级一下最新版本试试呢？Windows 系统版本是什么？有无相关日志？

**Reply by sxjsjt** [reply]
Windows是win11专业版22H2  内部版本号22621.3296

**Comment by QQQIANQ** [comment]
你好 昨天安装完最新版本的插件后 在刮削时 输入豆瓣id (比如输入：34937650 ) 后 无法刮削 当我查看日志时发现

[2024-05-24 09:30:28.936 +00:00] [ERR] Provider "Douban" failed to retrieve search results
System.FormatException: The input string '0.0' was not in a correct format.
   at System.Number.ThrowFormatException[TChar](ReadOnlySpan`1 value)
   at System.Number.ParseDecimal[TChar](ReadOnlySpan`1 value, NumberStyles styles, NumberFormatInfo info)
   at System.Convert.ToDecimal(String value)
   at Jellyfin.Plugin.Douban.Helper.ParseMovie(String responseText, String sid) in D:\a\JellyfinPluginDouban\JellyfinPluginDouban\Jellyfin.Plugin.Douban\Helper.cs:line 327
   at Jellyfin.Plugin.Douban.DoubanApi.FetchMovie(String sid, CancellationToken token) in D:\a\JellyfinPluginDouban\JellyfinPluginDouban\Jellyfin.Plugin.Douban\DoubanApi.cs:line 235
   at Jellyfin.Plugin.Douban.DoubanApi.GetMovieSearchResults(ItemLookupInfo info, Boolean tryParse, CancellationToken token) in D:\a\JellyfinPluginDouban\JellyfinPluginDouban\Jellyfin.Plugin.Douban\DoubanApi.cs:line 89
   at Jellyfin.Plugin.Douban.Provider.MovieProvider.GetSearchResults(MovieInfo info, CancellationToken token) in D:\a\JellyfinPluginDouban\JellyfinPluginDouban\Jellyfin.Plugin.Douban\Provider\MovieProvider.cs:line 52
   at MediaBrowser.Providers.Manager.ProviderManager.GetRemoteSearchResults[TItemType,TLookupType](RemoteSearchQuery`1 searchInfo, BaseItem referenceItem, CancellationToken cancellationToken)
   
  服务器版本ubuntu server 20.04
  jellyfin 版本 10.9.2

**Reply by Xzonn** [reply]
我这边好像复现不了，插件版本是最新的吗（目前是 2.2.0.0）？以及请发一下设置页面的截图（Cookie 可以去掉）。

**Reply by QQQIANQ** [reply]
在设置上好像没有什么特别的 有没有cookie 都没有任何改变 
<img width="832" alt="Capture d’écran 2024-05-24 à 18 37 34" src="https://github.com/Xzonn/xzonn.github.io/assets/33693109/260481f7-8b8c-4ea6-8618-768a50146393">
<img width="829" alt="Capture d’écran 2024-05-24 à 18 38 13" src="https://github.com/Xzonn/xzonn.github.io/assets/33693109/ba543ff2-e043-4135-89d9-74ade27fb81a">
版本是最新的 2.2.0.0

**Reply by Xzonn** [reply]
我这边在相同配置下是可以正常获取到元数据的，可能需要提供更详细的日志记录以便我排查问题。

![image](https://github.com/Xzonn/xzonn.github.io/assets/30619816/62550575-a4ad-4f01-8010-6735ae7b5bc0)

![image](https://github.com/Xzonn/xzonn.github.io/assets/30619816/18a7f6c0-3fc9-4282-a0fe-2d508ec6d8dc)

**Reply by Xzonn** [reply]
如果是 Ubuntu 系统的话，配置文件应该在 `/etc/jellyfin/` 文件夹下，在这个文件夹下新建一个文件，名为 `logging.json`，填入以下内容：

``` json
{
    "Serilog": {
        "MinimumLevel": {
            "Default": "Information",
            "Override": {
                "Microsoft": "Warning",
                "System": "Warning",
                "Jellyfin.Plugin.Douban": "Debug"
            }
        },
        "WriteTo": [
            {
                "Name": "Console",
                "Args": {
                    "outputTemplate": "[{Timestamp:HH:mm:ss}] [{Level:u3}] {Message:lj}{NewLine}{Exception}"
                }
            },
            {
                "Name": "Async",
                "Args": {
                    "configure": [
                        {
                            "Name": "File",
                            "Args": {
                                "path": "%JELLYFIN_LOG_DIR%//jellyfin.log",
                                "fileSizeLimitBytes": 10485700,
                                "rollOnFileSizeLimit": true,
                                "retainedFileCountLimit": 10,
                                "outputTemplate": "[{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz}] [{Level:u3}] {Message}{NewLine}{Exception}"
                            }
                        }
                    ]
                }
            }
        ]
    }
}
```

然后重启 Jellyfin，应该会记录本插件的详细日志，这样可以看出来是哪一步出了问题。

**Reply by QQQIANQ** [reply]
[log.txt](https://github.com/Xzonn/xzonn.github.io/files/15438592/log.txt)
这里是重启过后的详细日记

**Reply by QQQIANQ** [reply]
对了 我在插件中测试服务器是没问题的 可以正常显示图片

**Reply by QQQIANQ** [reply]
LANG=fr_FR.UTF-8
LANGUAGE=
LC_CTYPE="fr_FR.UTF-8"
LC_NUMERIC="fr_FR.UTF-8"
LC_TIME="fr_FR.UTF-8"
LC_COLLATE="fr_FR.UTF-8"
LC_MONETARY="fr_FR.UTF-8"
LC_MESSAGES="fr_FR.UTF-8"
LC_PAPER="fr_FR.UTF-8"
LC_NAME="fr_FR.UTF-8"
LC_ADDRESS="fr_FR.UTF-8"
LC_TELEPHONE="fr_FR.UTF-8"
LC_MEASUREMENT="fr_FR.UTF-8"
LC_IDENTIFICATION="fr_FR.UTF-8"
LC_ALL=

有一种可能 跟我的服务器地区设置有关系 我的服务器地区设置在法国
这可能会导致
在Helper.cs中
316    var rating = content?.QuerySelector("#interest_sectl .rating_num")?.InnerText.Trim();
317    rating = string.IsNullOrEmpty(rating) ? "0.0" : rating;
这里的rating在
353            Rating = Convert.ToDecimal(rating),
发生转换的时候产生格式异常
这也解释了日志中的"7.0" 因为7.0刚好是庆余年第二季在豆瓣上的评分

**Reply by QQQIANQ** [reply]
确实是这个原因 我把服务器的地区设置改成英语 就没有错误了 刮削也能够正常运行了
因为在法语中 评分是 "7,0" 而不是 "7.0" 英语国家就没有这个问题了
如果不介意的话 可以在下一个版本中加入使用不变的文化设置来转换评分 比如:
var parsedRating = decimal.Parse(rating, CultureInfo.InvariantCulture);
...
Rating = parsedRating,
仅供参考 但是这个bug挺有意思的

**Reply by Xzonn** [reply]
感谢测试，这倒确实挺有意思，我没考虑过小数点可能是逗号的情况。将在下个版本修复。

**Comment by FROKING** [comment]
希望能加入对豆瓣音乐元数据的刮削支持

**Comment by peikunkun** [comment]
操作都按照做了，点击测试服务器没反应了，前段时间还是可以出现豆瓣图片的

**Reply by peikunkun** [reply]
没事了，windows 客户端没有图片显示而已

**Comment by freely-boss** [comment]
豆瓣好像更新了，可以更新下吗，有些演职员的头像左下角有个“豆瓣字样”会导致刮不到这个人的图片，比如异人之下之决战碧游村，里面文淇和导演的照片就刮不到。

**Reply by Xzonn** [reply]
已修复。

**Comment by n0vemb** [comment]
要是能支持emby就好了

**Comment by freely-boss** [comment]
大佬，我发现背景图搜索，封面图搜索功能好像有bug，能搜出来，但是显示不了缩略图，也不能下载，我点击图片跳转提示403，默认全部是img2的地址了，我看了一下豆瓣图库那边是随机的，有img1，img2，img3，

**Reply by Xzonn** [reply]
参见 [#关于速率限制](https://xzonn.top/posts/Jellyfin-Plugin-Douban.html#%E5%85%B3%E4%BA%8E%E9%80%9F%E7%8E%87%E9%99%90%E5%88%B6)。

**Reply by freely-boss** [reply]
感谢，解决了，加了反代

**Comment by Bairiyuyu** [comment]
大佬，目前发现一个问题，以前都是可以刮削所有图片，现在 刮削影片可以刮削到封面图， 但是 徽标 和 背景图 无法刮削，目前都是空白，不知道是什么问题，可以抽空帮我看一下这个问题吗？谢谢。

目前我这边的配置有正确填写Cookie ，使用了自带的豆瓣图片服务器（如果不用的话，无法下载到图片）。其它均为默认设置。

**Reply by Xzonn** [reply]
豆瓣从来都不提供徽标图，你搜到的可能是别的插件提供的。背景图搜不到的话，在设置里把“根据宽高比区分海报和背景图”关掉。

**Reply by Bairiyuyu** [reply]
按照你的建议取消“根据宽高比区分海报和背景图”设置后，还是无法提取背景图，我印象中，就是忽然某一天起，就出现无法刮削徽标和背景图了。下面展示对比图。
![1](https://github.com/user-attachments/assets/b2b10dfa-01d3-46ce-ac40-fc7d17b3ecea)
![2](https://github.com/user-attachments/assets/23034524-8a0a-423b-8398-64cbfce9374e)
![3](https://github.com/user-attachments/assets/616f3601-d588-407f-ada9-7463ff162cc9)
![4](https://github.com/user-attachments/assets/ac64095e-252f-4dbe-a22a-430126956fa9)

**Reply by Xzonn** [reply]
@Bairiyuyu 更新一下插件版本，然后刷新元数据，或者点搜索图标手动添加。
